cilium/cmd/recorder_list.go

```diff
@@ -0,0 +1,135 @@
+)
+
+// recorderListCmd represents the recorder_list command
+var recorderListCmd = &cobra.Command{
+	Use:     "list",
+	Aliases: []string{"ls"},
+	Short:   "List current pcap recorders",
+	Run: func(cmd *cobra.Command, args []string) {
+		listRecorders(cmd, args)
+		listMasks(cmd, args)
+	},
+}
+
+func init() {
+	recorderCmd.AddCommand(recorderListCmd)
+	command.AddJSONOutput(recorderListCmd)
+}
+
+func listMasks(cmd *cobra.Command, args []string) {
+	list, err := client.GetRecorderMasks()
+	if err != nil {
+		Fatalf("Cannot get recorder mask list: %s", err)
+	}
+
+	if command.OutputJSON() {
+		if err := command.PrintOutput(list); err != nil {
+			os.Exit(1)
+		}
+		return
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 5, 0, 3, ' ', 0)
+	printRecorderMaskList(w, list)
+}
+
+func printRecorderMaskList(w *tabwriter.Writer, maskList []*models.RecorderMask) {
+	fmt.Fprintln(w, "Users\tPriority   \tWildcard Masks\t")
+	for _, mask := range maskList {
+		if mask.Status == nil || mask.Status.Realized == nil {
+			fmt.Fprint(os.Stderr, "error parsing recorder: empty state")
+			continue
+		}
+	}
+	sort.Slice(maskList, func(i, j int) bool {
+		return maskList[i].Status.Realized.Priority > maskList[j].Status.Realized.Priority
+	})
+	for _, mask := range maskList {
+		spec := mask.Status.Realized
+		str := fmt.Sprintf("%d\t%d\t%s:%s\t->\t%s:%s\t%s",
+			spec.Users, spec.Priority,
+			spec.SrcPrefixMask, spec.SrcPortMask,
+			spec.DstPrefixMask, spec.DstPortMask,
+			spec.ProtocolMask)
+		fmt.Fprintln(w, str)
+	}
+	w.Flush()
+}
+
+func listRecorders(cmd *cobra.Command, args []string) {
+	list, err := client.GetRecorder()
+	if err != nil {
+		Fatalf("Cannot get recorder list: %s", err)
+	}
+
+	if command.OutputJSON() {
+		if err := command.PrintOutput(list); err != nil {
+			os.Exit(1)
+		}
+		return
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 5, 0, 3, ' ', 0)
+	printRecorderList(w, list)
+}
+
+func printRecorderList(w *tabwriter.Writer, recList []*models.Recorder) {
+	fmt.Fprintln(w, "ID\tCapture Length\tWildcard Filters\t")
+	for _, rec := range recList {
+		if rec.Status == nil || rec.Status.Realized == nil {
+			fmt.Fprint(os.Stderr, "error parsing recorder: empty state")
+			continue
+		}
+	}
+	sort.Slice(recList, func(i, j int) bool {
+		return *recList[i].Status.Realized.ID <= *recList[j].Status.Realized.ID
+	})
+	for _, rec := range recList {
+		spec := rec.Status.Realized
+		capLen := "full"
+		if spec.CaptureLength != 0 {
+			capLen = fmt.Sprintf("<= %d", spec.CaptureLength)
+		}
+		str := fmt.Sprintf("%d\t%s\t%s:%s\t->\t%s:%s\t%s",
+			int64(*spec.ID), capLen,
+			spec.Filters[0].SrcPrefix, spec.Filters[0].SrcPort,
+			spec.Filters[0].DstPrefix, spec.Filters[0].DstPort,
+			spec.Filters[0].Protocol)
+		fmt.Fprintln(w, str)
+		for _, filter := range spec.Filters[1:] {
+			str := fmt.Sprintf("\t\t%s:%s\t->\t%s:%s\t%s",
+				filter.SrcPrefix, filter.SrcPort,
+				filter.DstPrefix, filter.DstPort,
+				filter.Protocol)
+			fmt.Fprintln(w, str)
+		}
```

Comment:
Do we need to check for `len(spec.Filters) > 0` here? I haven't seen any checks that stop a user from submitting a recording without filters.

Reply:
Actually all the code can handle it, e.g. I could reserve an empty id/caplen and then add/del filter later on.